File: examples/python/generate_video.py
"""
Sora API - Video Generation Example
Generate videos using Mountsea AI's Sora API

Documentation: https://docs.mountsea.ai/api-reference/sora/introduction
Platform: https://offoff.ai/
"""

import requests
import time
import os

API_KEY = os.environ.get("MOUNTSEA_API_KEY", "your-api-key")
BASE_URL = "https://api.mountsea.ai"

HEADERS = {
    "Authorization": f"Bearer {API_KEY}",
    "Content-Type": "application/json"
}


def text_to_video(prompt: str, duration: int = 5, resolution: str = "1080p") -> dict:
    """
    Generate a video from text description.
    
    Args:
        prompt: Description of the video to generate
        duration: Duration in seconds
        resolution: Video resolution (720p, 1080p, 4K)
    
    Returns:
        Task information including taskId
    """
    response = requests.post(
        f"{BASE_URL}/sora/generate",
        headers=HEADERS,
        json={
            "prompt": prompt,
            "duration": duration,
            "resolution": resolution
        }
    )
    response.raise_for_status()
    return response.json()


def image_to_video(prompt: str, image_url: str, duration: int = 5) -> dict:
    """
    Generate a video from an image.
    
    Args:
        prompt: Description of the animation/motion
        image_url: URL of the source image
        duration: Duration in seconds
    
    Returns:
        Task information including taskId
    """
    response = requests.post(
        f"{BASE_URL}/sora/generate",
        headers=HEADERS,
        json={
            "prompt": prompt,
            "imageUrl": image_url,
            "duration": duration
        }
    )
    response.raise_for_status()
    return response.json()


def wait_for_completion(task_id: str, timeout: int = 600, interval: int = 15) -> dict:
    """
    Wait for a video generation task to complete.
    
    Args:
        task_id: The task ID to monitor
        timeout: Maximum wait time in seconds
        interval: Polling interval in seconds
    
    Returns:
        Completed task results
    """
    start_time = time.time()
    while time.time() - start_time < timeout:
        response = requests.get(
            f"{BASE_URL}/sora/task",
            headers=HEADERS,
            params={"taskId": task_id}
        )
        result = response.json()
        status = result.get("status", "unknown")
        
        print(f"Status: {status}")
        
        if status == "completed":
            return result
        elif status == "failed":
            raise Exception(f"Task failed: {result.get('error', 'Unknown error')}")
        
        time.sleep(interval)
    
    raise TimeoutError(f"Task {task_id} did not complete within {timeout} seconds")


if __name__ == "__main__":
    # Example 1: Text-to-Video
    print("🎬 Generating video from text...")
    task = text_to_video(
        prompt="A golden retriever running through a field of sunflowers at sunset, cinematic, 4K",
        duration=5,
        resolution="1080p"
    )
    print(f"Task created: {task['taskId']}")
    
    print("⏳ Waiting for generation...")
    result = wait_for_completion(task['taskId'])
    print(f"✅ Video generated! URL: {result.get('videoUrl')}")
    
    # Example 2: Image-to-Video
    print("\n🖼️ Generating video from image...")
    task2 = image_to_video(
        prompt="The landscape slowly pans from left to right with clouds drifting",
        image_url="https://example.com/landscape.jpg",
        duration=5
    )
    print(f"Task created: {task2['taskId']}")
    
    print("\n🔗 Documentation: https://docs.mountsea.ai/api-reference/sora/introduction")
    print("🏠 Platform: https://offoff.ai/")

File: examples/php/generate_video.php
<?php
/**
 * Sora 2 API - Video Generation Example (PHP)
 * Generate videos using OpenAI Sora 2 via Mountsea AI
 *
 * Documentation: https://docs.mountsea.ai/api-reference/sora/introduction
 * Platform: https://offoff.ai/
 */

$API_KEY = getenv('MOUNTSEA_API_KEY') ?: 'your-api-key';
$BASE_URL = 'https://api.mountsea.ai';

/**
 * Generate video from text prompt (Sora 2 / Sora Pro)
 */
function generateVideo(string $prompt, int $duration = 5, string $resolution = '1080p'): array
{
    global $API_KEY, $BASE_URL;

    $ch = curl_init("$BASE_URL/sora/generate");
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_POST => true,
        CURLOPT_HTTPHEADER => [
            "Authorization: Bearer $API_KEY",
            'Content-Type: application/json',
        ],
        CURLOPT_POSTFIELDS => json_encode([
            'prompt' => $prompt,
            'duration' => $duration,
            'resolution' => $resolution,
        ]),
    ]);

    $response = curl_exec($ch);
    curl_close($ch);

    return json_decode($response, true);
}

/**
 * Check task status
 */
function getTaskStatus(string $taskId): array
{
    global $API_KEY, $BASE_URL;

    $ch = curl_init("$BASE_URL/sora/task?taskId=$taskId");
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_HTTPHEADER => [
            "Authorization: Bearer $API_KEY",
        ],
    ]);

    $response = curl_exec($ch);
    curl_close($ch);

    return json_decode($response, true);
}

/**
 * Wait for task completion
 */
function waitForCompletion(string $taskId, int $timeout = 600, int $interval = 10): array
{
    $start = time();
    while (time() - $start < $timeout) {
        $result = getTaskStatus($taskId);
        echo "Status: {$result['status']}\n";

        if ($result['status'] === 'completed') {
            return $result;
        }
        if ($result['status'] === 'failed') {
            throw new Exception("Task failed: " . ($result['error'] ?? 'Unknown'));
        }

        sleep($interval);
    }
    throw new Exception("Timeout waiting for task $taskId");
}

// --- Main ---
echo "🎬 Generating video with Sora 2...\n";

$task = generateVideo(
    'A golden retriever playing in autumn leaves, cinematic lighting, 4K',
    5,
    '1080p'
);
echo "Task ID: {$task['taskId']}\n";

echo "⏳ Waiting for completion...\n";
$result = waitForCompletion($task['taskId']);
echo "✅ Video URL: {$result['videoUrl']}\n";

echo "\n📘 Docs: https://docs.mountsea.ai/api-reference/sora/introduction\n";
echo "🏠 Platform: https://offoff.ai/\n";

